src/harness/testcases/WINNF_FT_S_HBT_testcase.py

```diff
@@ -0,0 +1,191 @@
+                                             '%Y-%m-%dT%H:%M:%SZ')
+    self.assertLessEqual(transmit_expire_time, datetime.utcnow())
+
+  @winnforum_testcase
+  def test_WINNF_FT_S_HBT_8(self):
+    """Heartbeat Array Request: BLACKLISTED.
+
+    Response Code should  be 101.
+    """
+    # Register the devices
+    registration_request = []
+    fcc_ids = []
+    for device_filename in ('device_a.json', 'device_c.json', 'device_e.json'):
+      device = json.load(
+          open(os.path.join('testcases', 'testdata', device_filename)))
+      fcc_ids.append(device['fccId'])
+      registration_request.append(device)
+
+    cbsd_ids = self.assertRegistered(registration_request)
+
+    # Request grant
+    grant_request = []
+    for cbsd_id in cbsd_ids:
+      grant = json.load(
+          open(os.path.join('testcases', 'testdata', 'grant_0.json')))
+      grant['cbsdId'] = cbsd_id
+      grant_request.append(grant)
+    request = {'grantRequest': grant_request}
+
+    # Check grant response
+    response = self._sas.Grant(request)['grantResponse']
+    self.assertEqual(len(response), len(cbsd_ids))
+    grant_ids = []
+    grant_expire_time = []
+    for resp_number, resp in enumerate(response):
+      self.assertEqual(resp['cbsdId'], cbsd_ids[resp_number])
+      self.assertTrue(resp['grantId'])
+      self.assertEqual(resp['response']['responseCode'], 0)
+      grant_ids.append(resp['grantId'])
+      grant_expire_time.append(
+          datetime.strptime(resp['grantExpireTime'], '%Y-%m-%dT%H:%M:%SZ'))
+    del request, response
+
+    # Inject Third Device into Blacklist
+    self._sas_admin.BlacklistByFccId({'fccId': fcc_ids[2]})
+
+    # First Heartbeat Request
+    heartbeat_request = []
+    for cbsd_id, grant_id in zip(cbsd_ids, grant_ids):
+      heartbeat_request.append({
+          'cbsdId': cbsd_id,
+          'grantId': grant_id,
+          'operationState': 'GRANTED'
+      })
+
+    request = {'heartbeatRequest': heartbeat_request}
+    response = self._sas.Heartbeat(request)['heartbeatResponse']
+    self.assertEqual(len(response), len(grant_ids))
+
+    # Check the heartbeat response
+    # First two devices have Response Code 0.
+    for index, resp in enumerate(response[:2]):
+      self.assertEqual(resp['cbsdId'], cbsd_ids[index])
+      self.assertEqual(resp['grantId'], grant_ids[index])
+      self.assertEqual(resp['response']['responseCode'], 0)
+      transmit_expire_time = datetime.strptime(resp['transmitExpireTime'],
+                                               '%Y-%m-%dT%H:%M:%SZ')
+      self.assertLess(datetime.utcnow(), transmit_expire_time)
+      self.assertLessEqual(
+          (transmit_expire_time - datetime.utcnow()).total_seconds(), 240)
+      self.assertLessEqual(transmit_expire_time,
+                           grant_expire_time[index])
+
+    # Third device is in Blacklist, must have Response Code 101.
+    self.assertEqual(response[2]['response']['responseCode'], 101)
```

Comment:
this line is ok according to the tests spec, but I don't know why here we don't allow response code 103, that we allow in the next step. this comment is not concerning the code of the test, but more the specification of the test.
what do you think?

Reply:
Reda, 
I understand this is regarding the specification. If you have any concerns with the test spec, you should file comments by tomorrow on the test spec.

Reply:
thank you